prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -147,86 +148,48 @@ func (af *GCSArtifactFetcher) artifacts(key string) ([]string, error) {
 	return artifacts, nil
 }
 
-const (
-	anonHost   = "storage.googleapis.com"
-	cookieHost = "storage.cloud.google.com"
-)
-
-func (af *GCSArtifactFetcher) signURL(bucket, obj string) (string, error) {
-	// We specifically want to use cookie auth, see:
-	// https://cloud.google.com/storage/docs/access-control/cookie-based-authentication
-	if af.useCookieAuth {
-		artifactLink := &url.URL{
-			Scheme: httpsScheme,
-			Host:   cookieHost,
-			Path:   path.Join(bucket, obj),
-		}
-		return artifactLink.String(), nil
-	}
-
-	// If we're anonymous we can just return a plain URL.
-	if af.gcsCredsFile == "" {
-		artifactLink := &url.URL{
-			Scheme: httpsScheme,
-			Host:   anonHost,
-			Path:   path.Join(bucket, obj),
-		}
-		return artifactLink.String(), nil
-	}
-
-	// TODO(fejta): do not require the json file https://github.com/kubernetes/test-infra/issues/16489
-	// As far as I can tell, there is no sane way to get these values other than just
-	// reading them out of the JSON file ourselves.
-	f, err := os.Open(af.gcsCredsFile)
-	if err != nil {
-		return "", err
-	}
-	defer f.Close()
-	auth := struct {
-		Type        string `json:"type"`
-		PrivateKey  string `json:"private_key"`
-		ClientEmail string `json:"client_email"`
-	}{}
-	if err := json.NewDecoder(f).Decode(&auth); err != nil {
-		return "", err
-	}
-	if auth.Type != "service_account" {
-		return "", fmt.Errorf("only service_account GCS auth is supported, got %q", auth.Type)
-	}
-	return storage.SignedURL(bucket, obj, &storage.SignedURLOptions{
-		Method:         "GET",
-		Expires:        time.Now().Add(10 * time.Minute),
-		GoogleAccessID: auth.ClientEmail,
-		PrivateKey:     []byte(auth.PrivateKey),
+func (af *GCSArtifactFetcher) signURL(ctx context.Context, key string) (string, error) {
+	return af.opener.SignedURL(ctx, key, pkgio.SignedURLOptions{
+		UseGSCookieAuth: af.useCookieAuth,
```

Comment:
Wouldn't it make more sense to pass this in when opening the opener? As far as I can see we either do this always or never or did I miss something?

Reply:
Let's continue here: https://github.com/kubernetes/test-infra/pull/17183#discussion_r415277033